File: AppcircleTests/AppcircleTests.m
//
//  AppcircleTests.m
//  AppcircleTests
//
//  Created by Dogan Ekici on 4.10.2019.
//  Copyright © 2019 Dogan Ekici. All rights reserved.
//

#import <XCTest/XCTest.h>

@interface AppcircleTests : XCTestCase

@end

@implementation AppcircleTests

- (void)setUp {
    // Put setup code here. This method is called before the invocation of each test method in the class.
}

- (void)tearDown {
    // Put teardown code here. This method is called after the invocation of each test method in the class.
}

- (void)testExample {
    // This is an example of a functional test case.
    // Use XCTAssert and related functions to verify your tests produce the correct results.
}

- (void)testArrayCount {
    NSArray *array = [[NSArray alloc] initWithObjects:@"First",@"Second", nil];
    XCTAssertEqual(array.count, 2);
}

- (void)testArrayFirstObject {
    NSArray *array = [[NSArray alloc] initWithObjects:@"First",@"Second", nil];
    XCTAssertEqual(array.firstObject, @"First");
}

- (void)testArrayLastObject {
    NSArray *array = [[NSArray alloc] initWithObjects:@"First",@"Second", nil];
    XCTAssertEqual(array.lastObject, @"Second");
}

- (void)testType {
    NSArray *array = [[NSArray alloc] initWithObjects:@"First",@"Second", nil];
    XCTAssert([array isKindOfClass:[NSArray class]]);
}

- (void)testPerformanceExample {
    // This is an example of a performance test case.
    [self measureBlock:^{
        // Put the code you want to measure the time of here.
    }];
}

@end

File: AppcircleTests/AppcircleTests.swift
//
//  AppcircleTests.swift
//  AppcircleTests
//
//  Created by Mustafa on 29.12.2021.
//

import XCTest
@testable import Appcircle

class AppcircleTests: XCTestCase {

    override func setUpWithError() throws {
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDownWithError() throws {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
    }

    func testSkipiPhone8() throws {
        let simulator = ProcessInfo.processInfo.environment["SIMULATOR_DEVICE_NAME"] ?? "No Name"
        try XCTSkipIf(simulator.contains("iPhone 8"), "Skip this test on iPhone 8*")
        XCTAssertTrue(true,"iPhone 8 Test")
    }
    
    func testSkipiPhone14() throws {
        let simulator = ProcessInfo.processInfo.environment["SIMULATOR_DEVICE_NAME"] ?? "No Name"
        try XCTSkipIf(simulator.contains("iPhone 14"), "Skip this test on iPhone 14*")
        XCTAssertTrue(true,"iPhone 14 Test")
    }

    
    func testExpectedFailure() throws {
        let thingThatFails: Bool = false
        XCTExpectFailure("Working on a fix for this problem.")
        XCTAssertTrue(thingThatFails, "This is not working right now.")
    }
        
    func testPass() throws {
        XCTAssertTrue(true, "This test should pass")
    }
    
    func testFail() throws {
        XCTAssertTrue(false,"This test should fail")
    }
    
    func testPerformanceExample() throws {
        // This is an example of a performance test case.
        self.measure {
            // Put the code you want to measure the time of here.
        }
    }

}
